### DISCUSSION TITLE
BitTimingFd validate questions

### DISCUSSION THREAD

**Discussion by Yuvalsh178** [hardbyte/python-can#1616] [General]
Hello,
I'm using the python-can 4.2.2/4.2.1 versions to integrate with Vector Devices, 

There are few BitTimingFd validations that are not clear for me. 

* can/bit_timing.py:536, if self.dbt < 8:
   According to Vector it seems that the BTL cycle can be less then 8.
   There is a reason that you limit this field to be <8 ? 
   When I removed your limitation (row 536,537) all CAN messages I sent were received 
  sucssefluuly.  

*  can/bit_timing.py:554: if not 25_000 <= self.data_bitrate <= 8_000_000:
    According to Vector the limitation should be 5K to 10M
   There is a reason that you limit this field to be: >25K and < 8M ?
 
*  can/bit_timing.py:571: if not 1 <= self.data_tseg1 <= 32:
    According to Vector this field can be greater than 32
   There is a reason that you limit this field

Appreciate your answer.
Thanks,
Yuval

**Comment by zariiii9003** [comment]
I deleted my earlier comment, you're right, the values are incorrect. For CAN FD the nominal bit time range shall be at least 8 to 80, and the data bit time range shall be at least 5 to 25.

Can you tell me which bit timings you are trying to set?

**Reply by zariiii9003** [reply]
The bit timing validation should probably be use case dependent. 
If the bit timings are calculated from sample points, then the limits of ISO 11898 should be enforced for best compatibility. If the bit timings are given by the user, then the class should only perform the most basic validation.

**Reply by Yuvalsh178** [reply]
Thank you for your response,
This is an example to a bit timing I'm trying to configure (custom timing):
f_clock=80000000, nom_brp=200, nom_tseg1=5, nom_tseg2=2, nom_sjw=1, data_brp=400, data_tseg1=2, data_tseg2=1, data_sjw=1;

Is the fix you suggested above can be made on the next version ?

**Reply by Yuvalsh178** [reply]
This is the configuration as it appears in VECTOR CANoe:

![image](https://github.com/hardbyte/python-can/assets/118765214/c15ead82-30a1-4613-a9db-f849b24439c8)

**Reply by zariiii9003** [reply]
Could you take a look at #1618 ? I added the `strict` argument which defaults to `False`.

**Reply by Yuvalsh178** [reply]
Thanks [zariiii9003](https://github.com/zariiii9003),
I reviewed your PR (#1618), it looks okay and it covers my needs.
I'm running my tests to check if everything is working well with your changes.
Until now all tests are pass (each test with different speed configurations).
I'll let you know if I'll find any issue.
Thank you so much for your support.
